src/tqec/templates/display.py

```diff
@@ -29,6 +30,88 @@ def display_template(
         print()
 
 
+def display_templates_ascii(
+    templates: ComposedTemplate,
+    plaquette_indices: ty.Sequence[int] | None = None,
+    h_space: int = 5,
+    v_space: int = 2,
+    corner_mark: str = "■",
+) -> None:
+    """Display the templates as an ASCII art (multi-line string).
+
+    Args:
+        templates: the ComposedTemplate instance to display.
+        plaquette_indices: the plaquette indices that are forwarded to the call
+            to `template.instantiate` to get the actual template representation.
+        h_space: horizontal space allocated to each plaquette location (for index and separation element)
+        v_space: vertical space between two rows of plaquettes (value = 1 + num additional lines)
+    """
+    if plaquette_indices is None:
+        plaquette_indices = tuple(range(1, templates.expected_plaquettes_number + 1))
+    if h_space < 4:
+        print(f"WARNING: horizontal space value {h_space} is not enough. Changed to default value 5.")
+        h_space = 5
+    if v_space < 2:
+        print(f"WARNING: vertical space value {v_space} is not enough. Changed to default value 2.")
+        v_space = 2
+    if len(corner_mark) != 1:
```

Comment:
This *should* work most of the time because Python `str::len` seems to behave as you expect on unicode characters, but I can pretty much guarantee that it will not work for all inputs (see for example [this thread](https://discuss.python.org/t/peculiar-behaviour-involving-unicode-len-s-0-len-s-0-lower-is-this-expected/24469/5)).

For the moment, and because this will likely never be an issue, I am fine not changing that portion of the code. I just wanted to note that length of unicode characters might not work as we expect in all cases.

Reply:
Thanks for pointing out these corner cases.
Below you are suggesting '+' instead of '■', I will commit your suggestions.
The choice of '■' was to provide a very visible corner mark.